### PR TITLE
updates to support export process to bpmn20 xml file with all form de…

### PR DESCRIPTION
**Pull request by simon998yang**
…finitions embeded (referenced form not resolved, so we can import all individual forms later on.  For deployment, we still export form with form reference form resolved)

### CODE REVIEW

File: modules/flowable-ui-modeler/flowable-ui-modeler-logic/src/main/java/org/flowable/app/service/api/ModelService.java [hunk 1]
```diff
@@ -51,7 +53,7 @@
     
     Map<String,byte[]> getDecisionTableDefinitionsForProcess(String modelId);
 
-    BpmnModel getBpmnModel(AbstractModel model, Map<String, Model> formMap, Map<String, Model> decisionTableMap);
+    BpmnModel getBpmnModel(AbstractModel model, Map<String, Model> formMap, Map<String, Model> decisionTableMap, boolean resolveFormReference);
```
Comment (ruiboy) [inline modules/flowable-ui-modeler/flowable-ui-modeler-logic/src/main/java/org/flowable/app/service/api/ModelService.java]: Should we be making breaking changes to the API of this class?  Wont that make it harder to upgrade flowable versions?   Shouldn't we keep the existing method and just have it call our new impl with boolean=false?

File: modules/flowable-ui-modeler/flowable-ui-modeler-logic/src/main/java/org/flowable/app/service/editor/FlowableModelQueryService.java [hunk 2]
```diff
@@ -145,7 +150,7 @@ public ResultListDataRepresentation getModelsToIncludeInAppDefinition() {
         return result;
     }
 
-    public ModelRepresentation importProcessModel(HttpServletRequest request, MultipartFile file) {
+    public Object importProcessModel(HttpServletRequest request, MultipartFile file, boolean overwrite) {
```
Comment (ruiboy) [inline modules/flowable-ui-modeler/flowable-ui-modeler-logic/src/main/java/org/flowable/app/service/editor/FlowableModelQueryService.java:153]: again, should be breaking the public API like this?

**Review by ruiboy — CHANGES_REQUESTED** [review]
A couple of comments within.

File: modules/flowable-ui-modeler/flowable-ui-modeler-logic/src/main/java/org/flowable/app/service/api/ModelService.java (hunk 1 above)
Comment (simon998yang) [reply modules/flowable-ui-modeler/flowable-ui-modeler-logic/src/main/java/org/flowable/app/service/api/ModelService.java]: right. added it back

File: modules/flowable-ui-modeler/flowable-ui-modeler-logic/src/main/java/org/flowable/app/service/editor/FlowableModelQueryService.java (hunk 2 above)
Comment (simon998yang) [reply modules/flowable-ui-modeler/flowable-ui-modeler-logic/src/main/java/org/flowable/app/service/editor/FlowableModelQueryService.java:153]: underlying problem here is that flowable return html for ajax call exception (which we can enhance it to return json like we did for gasp) , but flowable wrapping all exception simply in BadReqeustException. and cannot carry structural data except the exception message. Here I need to return something like {'existingProcess': 'pKey', 'existingForms': ['fKey1', 'fKey2]} to show a confirmation window for overwriting. 

I checked again in system builder and gasp , there seems no neat way to use exception to carry other structural information back to client except the error message?

Comment (ruiboy) [reply modules/flowable-ui-modeler/flowable-ui-modeler-logic/src/main/java/org/flowable/app/service/editor/FlowableModelQueryService.java:153]: okay -- leave that as is.   might be useful to add your comment to source code though ??

**Review by ruiboy — APPROVED** [review]

**Pull request merged**